#722 Documentation for the Shift Deadline plugin (#73)
Inbibo/shift_docs: docs/integration_resources/software/deadline.md

Reviewer: Can you make this impersonal? Not using the pronoun "you"

Author: Fone

# Deadline

> Shift integration in Deadline is in the works. The Shift plugin for Deadline is currently only available on Windows and supports execution of Shift workflows in batch mode or within Maya.

Reviewer: Given that we are already documenting Deadline, is it necessary to keep "Shift integration in Deadline is in the works?

```suggestion
Shift integration in Deadline is in the works. The Shift plugin for Deadline is currently only available on Windows and supports the execution of Shift workflows in batch mode or within Maya.

```

Author: Yes, we should mention that it is in the works, right now we have the plugin so we can send a Shift workflow to Deadline. But we have no way to send any other jobs to deadline from Shift for the moment

# Shift Plugin for Deadline Renderfarm

This guide explains how to set up the Shift plugin for Deadline. The setup process varies depending on the version of Python used by the Deadline installation.

## Prerequisites

- Access to the Deadline repository.
- Administrative rights to create directories and modify settings in the Deadline Monitor.

> - The Shift software installed.

Reviewer: ```suggestion
- The Shift software installed and activated.

```

## Step 1: Create the Plugin Folder

1. Navigate to the Deadline repository at `<path_to_deadline_repository>/custom/plugins`.
2. Create a new folder named `Shift`.

## Step 2: Copy the Plugin Files

1. Locate the `deadline` folder in the Shift installation directory.  
   The path is `<path_to_the_shift_installation_folder>/deadline`.
2. Inside the `deadline` folder, there are two directories:

>    - `python2.7`

Reviewer: We should remove python 2.7 right?

Author: No, Shift uses python 3. But some companies might still have their Deadline working with python 2

   - `python3`
   
   Depending on the Python version used in the Deadline installation:
   - For **Python 2.7**, copy the contents of the `python2.7` folder.
   - For **Python 3**, copy the contents of the `python3` folder.

3. Paste the copied files into the `Shift` folder created in the Deadline repository.

## Step 3: Configure the Plugin in Deadline

1. Open **Deadline Monitor**.
2. Navigate to **Tools > Configure Plugins**.
3. Select `Shift` from the list of plugins for configuration.
4. In the `Miscellaneous` tab, configure the executable paths:
   - **Shift**: Set the path to the `shift_batch.bat` file, located in the root of the Shift installation directory.
   - **Shift Maya**: Set the path to the `shift_maya_batch.bat` file, also located in the root of the Shift installation directory.

## Step 4: Configuring Environment Variables (Optional)

To set environment variables for Shift while running on Deadline, two options are available:

### Option 1: Modify `PluginPreLoad.py`

1. Open the `PluginPreLoad.py` script from `<path_to_deadline_repository>/custom/plugins/Shift`.
2. Add or modify any necessary environment variables.

### Option 2: Use a Custom PreLoad Script

> A custom Python file can be created to set up environment variables, utilizing the following environment variable:

Reviewer 2: ```suggestion
A custom Python file can be created to set up environment variables, using the following environment variables:
```
